From 377cf1b720bf3c74a08251259a5c56474df20835 Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Mon, 23 Jun 2014 21:42:03 -0400
Subject: [scsi] lpfc 10.2.8001.0: Fix for initializing RRQ bitmap

Message-id: <1403559726-20488-22-git-send-email-revers@redhat.com>
Patchwork-id: 84441
O-Subject: [RHEL7.1 e-stor PATCH 21/24] scsi: lpfc 10.2.8001.0: Fix for initializing RRQ bitmap
Bugzilla: 1088574
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Dick Kennedy <rkennedy@Emulex.Com>

Submitted by James Smart James.Smart@emulex.com

Fix for initializing RRQ bitmap

Signed-off-by: James Smart james.smart@emulex.com

http://marc.info/?l=linux-scsi&m=140067391918967&w=2
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_hbadisc.c b/drivers/scsi/lpfc/lpfc_hbadisc.c
index a44d76a..d17cde6 100644
--- a/drivers/scsi/lpfc/lpfc_hbadisc.c
+++ b/drivers/scsi/lpfc/lpfc_hbadisc.c
@@ -5634,6 +5634,9 @@ lpfc_nlp_init(struct lpfc_vport *vport, struct lpfc_nodelist *ndlp,
   ndlp->active_rrqs_xri_bitmap =
     mempool_alloc(vport->phba->active_rrq_pool,
            GFP_KERNEL);
+  if (ndlp->active_rrqs_xri_bitmap)
+   memset(ndlp->active_rrqs_xri_bitmap, 0,
+          ndlp->phba->cfg_rrq_xri_bitmap_sz);
  }
 
 
-- 
1.7.1